fitlins/interfaces/afni.py

```diff
@@ -0,0 +1,627 @@
+        img = nb.load(img_path)
+
+        mask_file = self.inputs.mask_file
+        if not isdefined(mask_file):
+            mask_file = None
+        smoothing_fwhm = self.inputs.smoothing_fwhm
+        if not isdefined(smoothing_fwhm):
+            smoothing_fwhm = None
+
+        # Signal scaling occurs by default (the
+        # nistats.first_level_model.FirstLevelModel class rewrites the default
+        # signal_scaling argument of 0 to be True and then sets the
+        # scaling_axis attribute to 0
+        signal_scaling = True
+        scaling_axis = 0
+
+        # Since this estimator uses a 4d image instead of a 2d matrix the
+        # dataset axes must be mapped:
+        axis_mapping = {
+            # mean scaling each voxel with respect to time
+            0: (-1),
+            # mean scaling each time point with respect to all voxels
+            1: (0, 1, 2),
+            # scaling with respect to voxels  and time, which is known as grand mean scaling
+            (0, 1): None,
+        }
+        if signal_scaling:
+            img_mat = img.get_fdata()
+            mean = img_mat.mean(axis=axis_mapping[scaling_axis], keepdims=True)
+            if (mean == 0).any():
+                logger.warning(
+                    "Mean values of 0 observed."
+                    "The data have probably been centered."
+                    "Scaling might not work as expected"
+                )
+            mean = np.maximum(mean, 1)
+            img_mat = 100 * (img_mat / mean - 1)
+            img = type(img)(img_mat, img.affine)
+            img_path = "_scaled.".join(img_path.split("/")[-1].split(".", 1))
+            img.to_filename(img_path)
+
+        # Execute commands
+        logger.info(
+            f"3dREMLfit and 3dPval computation will be performed in: {tmpdir}\n"
+        )
+
+        # Define 3dREMLfit command
+        remlfit = afni.Remlfit()
+        remlfit.inputs.in_files = img_path
+        remlfit.inputs.matrix = design_fname
+        remlfit.inputs.out_file = "glt_results.nii.gz"
+        remlfit.inputs.var_file = "glt_extra_variables.nii.gz"
+        remlfit.inputs.wherr_file = "wherrorts.nii.gz"
+        remlfit.inputs.tout = True
+        remlfit.inputs.rout = True
+        remlfit.inputs.fout = True
+        remlfit.inputs.verb = True
+        remlfit.inputs.usetemp = True
+        remlfit.inputs.goforit = True
+        remlfit.inputs.mask = self.inputs.mask_file
+        reml_res = remlfit.run()
+
+        # calc smoothness
+        fwhm = afni.FWHMx()
+        fwhm.inputs.in_file = reml_res.outputs.wherr_file
+        fwhm.inputs.out_file = fname_fmt("model", "residsmoothness").replace('.nii.gz', '.tsv')
+        fwhm_res = fwhm.run()
+        fwhm_dat = pd.read_csv(fwhm_res.outputs.out_file,  delim_whitespace=True, header=None)
+        fwhm_dat.to_csv(fwhm_res.outputs.out_file, index=None, header=False, sep='\t')
+
+        #self._results['residual_smoothness'] = fwhm_res.outputs.out_file
```

Comment:
If we're going to have quoted code, can you do something like:

```Python
# XXX: When XYZ happens, we can uncomment this
# <code>
```

Reply:
@Shotgunosine, are you happy to remove this?

Reply:
Yep, that's fine to get rid of.

Reply:
I'll get rid of the rest of the commented out code as well.